src/transformState.js

```diff
@@ -5,7 +5,35 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case 'clear': {
+        if (action.type === 'clear') {
+          for (const key in state) {
+            delete state[key];
+          }
+        }
+
+        break;
+      }
+
+      case 'addProperties': {
+        Object.assign(state, action.extraData);
+
+        break;
+      }
+
+      case 'removeProperties': {
+        for (const elem of action.keysToRemove) {
+          if (state.hasOwnProperty(elem)) {
+            delete state[elem];
+          }
```

Comment:
you can delete the condition. If elem exists in the state, it will remove. If not - not

```suggestion
          delete state[elem];
```